src/services/services.ts

```diff
@@ -1086,6 +1088,12 @@ namespace ts {
         newText: string;
     }
 
+    export interface TextInsertion {
+        newText: string;
+        /** The position in newText the caret should point to after the insertion. */
+        caretOffset: number;

```

Comment:
I suppose I just don't see how those functionalities are relevant to this PR or feature. 

Do you want us to emit something like
```ts
/**
 * <+function_description+>
 * @param a <+param_description+>
 * @param b <+param_description+>
 */
```

and then we can iterate between them by tabbing or some hotkey (eg: <C-j>)?

(My understanding of the functionality is inspired by the Latex-Suite plugin for vim)